### PR TITLE
[DOCS] Added partition table tutorial

### LINKED ISSUE #4551: NEED help in setting up partition Scheme/Table for ESP32-WROOM-32D (8MB)
Am trying to find partition table for enabling partition scheme with boards.txt.
Am using a 8MB flash version of ESP32-WROOM-32D from digikey.
https://www.digikey.in/product-detail/en/espressif-systems/ESP32-WROOM-32D-8MB/1904-1024-6-ND/9381749

I am trying to increase the APP size to 4MB and SPIFFS to 3MB without OTA.

how to create the partition table for the 8MB FLASH Module.

Please Someone guide me.

I have arrived at these lines for scheme. will it work?

```
# Name,Type,SubType,Offset,Size,Flags
nvs,      data,  nvs,     0x9000,    0x5000	,
phy_init, data,  phy,     0xe000,    0x2000	,
factory,  app,   factory, 0x10000,   0x3500000	,
spiffs,   data,  spiffs,  0x3510000,  0xF0000	,

```

### LINKED ISSUE #6495: Partition Table/Option for 16MB flash (and 32MB flash)
### Related area

Board Support

### Hardware specification

ESP32-WROVER-B

### Is your feature request related to a problem?

I'm trying to flash my wrover-b with a 16MB flash partition, but i can't find any support at all for that. There is a option for ` (2mb app/12.5mb fat)` but i'm using LittleFS, so that won't work. I'm currently stuck using No OTA (1mb app/ 3mb spiffs) which works, but only allows for 25% of my total capacity. Additionally, we have a order of chips coming in soon that will have 32MB total capacity (256Mb) of NOR flash mem. so it'll be even smaller of a ratio

### Describe the solution you'd like

A custom board template, partition table, or just a start point for someone completely new to the world of embedded systems. I've googled so much for an answer before coming here as this is a last  resort of sorts. I'd make the custom table, but don't really understand the mem address or size on the custom tables i've  seen.

Thanks in advance!

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

### PR DESCRIPTION
**Pull request by pedrominatel**
## Description of Change

This PR adds the partition table details and the steps for creating a custom one.

## Tests scenarios

N/A

## Related links

Closes #4551
Closes #6495

### CODE REVIEW

File: docs/source/tutorials/partition_table.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
+###############
+Partition Table
+###############
+
+Introduction
+------------
+
+Partition table is used to define the flash memory organization and the different kind of data will be stored on each partition.
+
+You can use one of the predefined partition table schemes available or create your own.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:10]: You can use one of the **available**(moved) predefined partition table schemes or create your own.

File: docs/source/tutorials/partition_table.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+The partition table is created using a CSV file format with the following format:
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:12]: following **structure**?

File: docs/source/tutorials/partition_table.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+.. code-block::
+
+    # ESP-IDF Partition Table
+    # Name, Type, SubType, Offset, Size, Flags
+
+Where:
+
+1. **Name**
+
+    Is the partition name and must be a unique name. This name is not relevant for the system and the size must be at maximum of 16-chars.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:23]: Maybe mention to not use special characters?

File: docs/source/tutorials/partition_table.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    * ``app`` type is used to define the partition that will store the application.
+
+    * ``data`` type can be used to define the partition that stores general data, not the application.
+
+3. **SubType**
+
+    The SubType defines the usage of the ``app`` and ``data`` partitions.
+
+   **data**
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:37]: Are some of subtypes mandatory? I see some are optional (maybe we can highlight this more?)


_This is maybe not valid questions as more experienced Devs are going to add their custom partition table scheme._

File: docs/source/tutorials/partition_table.rst [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+5. **Size**
+
+    Size defines the amount of memory to be allocated on the partition. The size can be formatted as decimal, hex numbers (0x prefix), or using unit prefix K (kilo) or M (mega) i.e: 4096 = 4K = 0x1000.
+
+6. **Flags**
+
+    The last column in the CSV file is the flags and it is currently used to define if the partition will be encrypted by the flash encryption feature.
+
+
+For example, the most common partition is the ``default_8MB.csv`` (see `tools/partitions <https://github.com/espressif/arduino-esp32/tree/master/tools/partitions>`_ folder for some examples):
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:105]: use bold text for this?

File: docs/source/tutorials/partition_table.rst [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    nvs,      data, nvs,     0x9000,  0x5000,
+    otadata,  data, ota,     0xe000,  0x2000,
+    app0,     app,  ota_0,   0x10000, 0x330000,
+    app1,     app,  ota_1,   0x340000,0x330000,
+    spiffs,   data, spiffs,  0x670000,0x190000,
+
+Using a Custom Partition Scheme
+-------------------------------
+
+To create your own partition table, you can create the ``partitions.csv`` file in the same folder you created your sketch. The build system will automatically pick the partition table file and use it instead of the predefined ones.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/partition_table.rst:119]: bold text here as well? It's important info.

**Review by VojtechBartoska — APPROVED** [review]
Nice work @pedrominatel!

Left few comments (just suggestion to be added up to your opinion).

File: docs/source/tutorials/partition_table.rst (hunk 4 above)
Comment (pedrominatel) [reply docs/source/tutorials/partition_table.rst:37]: It will depend if you need OTA or not.

**Comment by pedrominatel** [comment]
PTAL @VojtechBartoska

**Comment by VojtechBartoska** [comment]
LGTM!

**Pull request merged**